### PR TITLE
Add support for `Stripe-Should-Retry` header

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @brandur-stripe @rattrayalex-stripe 
cc @stripe/api-libraries 

Add support for `Stripe-Should-Retry` header.

No tests because .NET _really_ doesn't want you messing with response headers for some mysterious reason. :(

### CODE REVIEW

File: src/StripeTests/Functional/NetworkRetriesTest.cs [hunk 1]
```diff
@@ -27,6 +28,11 @@ public NetworkRetriesTest(MockHttpClientFixture mockHttpClientFixture)
 
         private new IStripeClient StripeClient { get; }
 
+        // We really should be testing the behavior when the response has a `Stripe-Should-Retry`
+        // header, however .NET makes it basically impossible to test this because:
+        // 1. `HttpResponseMessage` doesn't let you set `Headers`, and
+        // 2. `HttpResponseHeaders` is a sealed class with no public constructor.
```
Comment (brandur-stripe) [inline src/StripeTests/Functional/NetworkRetriesTest.cs]: That's brutal :/

**Comment by brandur-stripe** [comment]
LGTM too @ob-stripe!

File: src/StripeTests/Functional/NetworkRetriesTest.cs (hunk 1 above)
Comment (ob-stripe) [reply src/StripeTests/Functional/NetworkRetriesTest.cs]: Indeed 😭

**Comment by ob-stripe** [comment]
Self-reapproving after fixing the linting issue.

Going to merge and release this, but cc @joek-stripe in case you want to flex some .NET muscle and write a test for this :p

**Comment by ob-stripe** [comment]
Released as 29.2.1.

**Pull request merged**